substra/cli/interface.py

```diff
@@ -341,9 +351,10 @@ def add_data_sample(ctx, path, dataset_key, local, multiple, test_only):
                 metavar="PATH")
 @click.option('--objective-key')
 @click_global_conf_with_output_format
+@click_option_metadata
 @click.pass_context
 @error_printer
-def add_dataset(ctx, data, objective_key):
+def add_dataset(ctx, data, objective_key, metadata):
```

Comment:
I think it would be better to not add an additional path but add a metadata field in the existing data file (same for all assets).